session/session.go

```diff
@@ -3230,12 +3230,16 @@ func (s *session) ShowProcess() *util.ProcessInfo {
 }
 
 // GetStartTSFromSession returns the startTS in the session `se`
-func GetStartTSFromSession(se interface{}) uint64 {
-	var startTS uint64
+func GetStartTSFromSession(se interface{}) (uint64, uint64) {
+	var startTS, processInfoID uint64
 	tmp, ok := se.(*session)
 	if !ok {
 		logutil.BgLogger().Error("GetStartTSFromSession failed, can't transform to session struct")
-		return 0
+		return 0, 0
+	}
+	processInfo := tmp.ShowProcess()
```

Comment:
we can get the id from "tmp.sessionVars.ConnectionID"

Reply:
As the analyze identifier is injected using `ProcessID` perhaps it's safer to use `ProcessID` to do the comparison.

Reply:
AutoAnalyzeProcID is assigned to `session.GetSessionVars().ConnectionID` and then `session.processInfo.ID` is set to `session.GetSessionVars().ConnectionID`.